addons/web/static/tests/setup.js

```diff
@@ -171,8 +173,21 @@ function patchBrowserWithCleanup() {
             localStorage: makeRAMLocalStorage(),
             sessionStorage: makeRAMLocalStorage(),
             // Don't want original animation frames in tests
-            requestAnimationFrame: (fn) => fn(),
-            cancelAnimationFrame: () => {},
+            requestAnimationFrame: (fn) => {
+                const handle = nextAnimationFrameHandle++;
+                animationFrameHandles.add(handle);
+
+                Promise.resolve().then(() => {
+                    if (animationFrameHandles.has(handle)) {
+                        fn(16);
+                    }
+                });
+
+                return handle;
+            },
+            cancelAnimationFrame: (handle) => {
+                animationFrameHandles.delete(handle);
+            },
```

Comment:
I hope this doesn't introduce sources of non deterministic errors on runbot. Was this strictly necessary? I thought we talked about only patching requestAnimationFrame in specific tests.